File: easy/26.remove-duplicates-from-sorted-array.py
"""
0,1,2,2,3
^ ^
0,1,0,1,1,2,2,3,3,4
  ^   ^         ^ ^   
move right until new number
swap left+1 and right
move left and right

[1,1,1,2]
   ^ ^
"""
class Solution:
    def removeDuplicates(self, nums: List[int]) -> int:
        nextnondub = 1
        for i in range(1,len(nums)): #0
            if nums[nextnondub-1] != nums[i]:
                nums[nextnondub] = nums[i]
                nextnondub += 1
        return nextnondub

            
        

File: easy/26.remove-duplicates-from-sorted-array.go
package main
func removeDuplicates(nums []int) int {
    n := 0
    for i, x := range nums {
        if i == 0 {
            nums[n] = x
		    n++
            continue
        }
	    if x != nums[n-1] {
		    nums[n] = x
		    n++
	    }
    }
    return n
    
}